site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx

```diff
@@ -92,258 +101,220 @@ export const CreateTemplateForm: FC<CreateTemplateFormProps> = ({
+                    </HelpTooltipText>
+                  </HelpTooltip>
                 </Stack>
+                <span className={styles.optionHelperText}>
+                  {t("form.helperText.allowUsersToCancel")}
+                </span>
               </Stack>
-            </label>
-          </Stack>
-        </div>
-
-        {/* Parameters */}
-        {parameters && (
-          <div className={styles.formSection}>
-            <div className={styles.formSectionInfo}>
-              <h2 className={styles.formSectionInfoTitle}>
-                {t("form.parameters.title")}
-              </h2>
-              <p className={styles.formSectionInfoDescription}>
-                {t("form.parameters.description")}
-              </p>
-            </div>
-
-            <Stack direction="column" className={styles.formSectionFields}>
-              {parameters.map((schema) => (
-                <ParameterInput
-                  schema={schema}
-                  disabled={isSubmitting}
-                  key={schema.id}
-                  onChange={async (value) => {
-                    await form.setFieldValue(
-                      `parameter_values_by_name.${schema.name}`,
-                      value,
-                    )
-                  }}
-                />
-              ))}
             </Stack>
-          </div>
-        )}
-
-        {jobError && (
-          <Stack>
-            <div className={styles.error}>
-              <h5 className={styles.errorTitle}>Error during provisioning</h5>
-              <p className={styles.errorDescription}>
-                Looks like we found an error during the template provisioning.
-                You can see the logs bellow.
-              </p>
+          </label>
+        </FormFields>
+      </FormSection>
+
+      {/* Parameters */}
+      {parameters && (
+        <FormSection
+          title={t("form.parameters.title")}
+          description={t("form.parameters.description")}
+        >
+          <FormFields>
+            {parameters.map((schema) => (
+              <ParameterInput
+                schema={schema}
+                disabled={isSubmitting}
+                key={schema.id}
+                onChange={async (value) => {
+                  await form.setFieldValue(
+                    `parameter_values_by_name.${schema.name}`,
+                    value,
+                  )
+                }}
+              />
+            ))}
+          </FormFields>
+        </FormSection>
+      )}
+
+      {/* Variables */}
+      {variables && (
+        <FormSection title="Variables" description="Template variables">
+          <FormFields>
+            {variables.map((variable, index) => (
+              <VariableInput
+                variable={variable}
+                disabled={isSubmitting}
+                key={variable.name}
+                onChange={async (value) => {
+                  await form.setFieldValue("user_variable_values." + index, {
```

Comment:
out of curiosity: why async/await?

Reply:
This is from the Formik library and for some reason, they made this function async. I think it is strange as well.

Reply:
TIL!